test/e2e/marketplace_test.go

```diff
@@ -0,0 +1,186 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	apis "github.com/operator-framework/operator-marketplace/pkg/apis"
+	operator "github.com/operator-framework/operator-marketplace/pkg/apis/marketplace/v1alpha1"
+
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/operator-framework/operator-sdk/pkg/test/e2eutil"
+
+	olm "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+var (
+	retryInterval        = time.Second * 5
+	timeout              = time.Second * 60
+	cleanupRetryInterval = time.Second * 1
+	cleanupTimeout       = time.Second * 5
+	testNamespace        = "operator-test"
+)
+
+func TestMarketplace(t *testing.T) {
+	operatorsourceList := &operator.OperatorSource{
+		TypeMeta: metav1.TypeMeta{
+			Kind: operator.OperatorSourceKind,
+			APIVersion: fmt.Sprintf("%s/%s",
+				operator.SchemeGroupVersion.Group, operator.SchemeGroupVersion.Version),
+		},
+	}
+	catalogsourceconfigList := &operator.CatalogSourceConfig{
+		TypeMeta: metav1.TypeMeta{
+			Kind: operator.CatalogSourceConfigKind,
+			APIVersion: fmt.Sprintf("%s/%s",
+				operator.SchemeGroupVersion.Group, operator.SchemeGroupVersion.Version),
+		},
+	}
+	err := framework.AddToFrameworkScheme(apis.AddToScheme, operatorsourceList)
+	if err != nil {
+		t.Fatalf("failed to add operatorsource custom resource scheme to framework: %v", err)
+	}
+	err = framework.AddToFrameworkScheme(apis.AddToScheme, catalogsourceconfigList)
+	if err != nil {
+		t.Fatalf("failed to add catalogsourceconfig custom resource scheme to framework: %v", err)
+	}
+	// run subtests
+	t.Run("marketplace-group", func(t *testing.T) {
+		t.Run("Cluster", MarketplaceCluster)
+	})
+}
+
+func defaultCreateTest(t *testing.T, f *framework.Framework, ctx *framework.TestCtx) error {
+	namespace, err := ctx.GetNamespace()
+	if err != nil {
+		return fmt.Errorf("could not get namespace: %v", err)
+	}
+
+	testOperatorSource := &operator.OperatorSource{
```

Comment:
Is there a way to use the example CR here?

Reply:
Afaik the only way for us to do that would be to parse it and pass it along. The client in the test framework doesn't have the capability to read yaml files.